pkg/apis/machine/validation/alicloudmachineclass.go

```diff
@@ -63,6 +64,34 @@ func validateAlicloudMachineClassSpec(spec *machine.AlicloudMachineClassSpec, fl
 		allErrs = append(allErrs, field.Required(fldPath.Child("keyPairName"), "KeyPairName is required"))
 	}
 
+	if spec.DataDisks != nil {
+		names := map[string]int{}
+		for i, dataDisk := range spec.DataDisks {
+			idxPath := fldPath.Child("dataDisks").Index(i)
+			if _, keyExist := names[dataDisk.Name]; keyExist {
+				names[dataDisk.Name]++
+			} else {
+				names[dataDisk.Name] = 1
+			}
+
+			if dataDisk.Name == "" {
```

Comment:
Also need to check Name string pattern. ```The name must be 2 to 128 characters in length, and can contain letters, digits, periods (.), underscores (_), colons (:), and hyphens (-). It must start with a letter.``` Should be better checked when the shoot is created, btw.

Reply:
The actual name is generated from the device name, instance name and some suffix. The machine name is safe, but I can't assume much about its length. To be a bit safer, I can enforce that data disk name length should be <64. Regarding the shoot validation - I can add it in the ac-provider extension. WDYT?

Reply:
Sure. Thank you.

Reply:
length and regex validation on the device name was added.